Corrected event subscriptions

straccio opened the pull request:
Corrections for event subscriptions. This change need this commit in
spark-protocol
https://github.com/straccio/spark-protocol/commit/0cc9a5a266834c56805861
f62e99015e3ddc83bb

dmiddlecamp commented:
Hi @straccio,

  Thanks for the PR!  If you haven't already, could you please sign the Contributors License Agreement?

https://docs.google.com/forms/d/e/1FAIpQLScvu-08mbw0CFSf9V0Xb2QaFn2pWNLGQy5mZcT0FZ2UnSSBTw/viewform

Thanks!
David

jlkalberer committed:
Trying to fix packages when installing this package.

jlkalberer committed:
Trying to fix npm install

jlkalberer committed:
Added exports so this library can be used as a npm package.

jlkalberer committed:
Fixing path for export.

jlkalberer committed:
Added dist folder so this repo can be consumed by other repos.

jlkalberer committed:
Fixing build... some comments broke node.

jlkalberer committed:
Updating bindings to work with external settings.

jlkalberer committed:
Passing settings into spark protocol

jlkalberer committed:
More work with settings

jlkalberer committed:
Update raspberryPi.md

jlkalberer committed:
Update raspberryPi.md

AntonPuko committed:
change passwordHasher function to sha256.

jlkalberer committed:
Merge pull request #123 from AntonPuko/dev

change passwordHasher function to sha256.

lbt committed:
Add Brewskey specific instructions and notes on babel.

Signed-off-by: David Greaves <david.greaves@jolla.com>

lbt committed:
Additional documentation to help get started with using or migrating to the Brewskey clone.

Signed-off-by: David Greaves <david.greaves@jolla.com>

jlkalberer committed:
Merge pull request #125 from lbt/brewskey

Getting started and minor docs update

jlkalberer committed:
Merge pull request #126 from Brewskey/dev

Bug fixes

jlkalberer committed:
Merge pull request #127 from Brewskey/master

Master

jlkalberer committed:
Working on integrating the electron

jlkalberer committed:
Merge branch 'master' of github.com:Brewskey/spark-server into dev

AntonPuko committed:
fix settings. add tcp_server_config: ENABLE_SYSTEM_FIRMWARE_AUTOUPDATES

AntonPuko committed:
move ENABLE_SYSTEM_FIRMWARE_AUTOUPDATES outside tcp server config

jlkalberer committed:
Merge pull request #128 from AntonPuko/dev

fix settings. add ENABLE_SYSTEM_FIRWMARE_AUTOUPDATES

AntonPuko committed:
transform webhookResponse buffer chunks to string

jlkalberer committed:
Merge pull request #129 from AntonPuko/dev

transform webhookResponse buffer chunks to string

suhajdab committed:
fix typo in install instructions

jlkalberer committed:
Merge pull request #130 from suhajdab/patch-1

fix typo in install instructions

jlkalberer committed:
Merge branch 'dev' of github.com:Brewskey/spark-server into dev

AntonPuko committed:
fix route config to prevent sent headers after submit error on particle subscribe calls

AntonPuko committed:
fix tab

AntonPuko committed:
Merge pull request #132 from AntonPuko/fixSentHeaders

fix route config to prevent sent headers after submit error on partic…

jlkalberer committed:
Updated start:prod to use the correct folder.

jlkalberer committed:
Fixed #134

AntonPuko committed:
add ssl support

jlkalberer committed:
Merge pull request #139 from AntonPuko/ssl

add ssl support

AntonPuko committed:
add tingodb, implement database repositories

jlkalberer committed:
Merge pull request #141 from AntonPuko/tingodb

add tingodb, implement database repositories

AntonPuko committed:
implement deviceAttributeDatabase repo

jlkalberer committed:
Merge pull request #142 from AntonPuko/attrDatabaseRepo

implement deviceAttributeDatabase repo

jlkalberer committed:
Removing postinstall as it breaks when you consume spark-server in other projects.

jlkalberer committed:
Fixing package.json

jlkalberer committed:
Updating exports so I can use the promisify code.

jlkalberer committed:
Changed the interface for using mongo

jlkalberer committed:
Added a mongo connector. Did some cleanup

AntonPuko committed:
made it work for tingo and mongo

jlkalberer committed:
Merge pull request #144 from AntonPuko/RepoCleanup

made it work for tingo and mongo

jlkalberer committed:
Allow users to override express config.

jlkalberer committed:
nits

AntonPuko committed:
add migrateScript

AntonPuko committed:
fix main.js Flow errors

jlkalberer committed:
Merge pull request #146 from AntonPuko/migrateScript

add migrateScript

jlkalberer committed:
Merge pull request #147 from AntonPuko/mainFix

fix main.js Flow errors

jlkalberer committed:
Added extra logging

jlkalberer committed:
Added webhook logger which can be overridden - closes #57

AntonPuko committed:
add interfaces for repositories

jlkalberer committed:
Merge pull request #151 from AntonPuko/interfaces

add interfaces for repositories

AntonPuko committed:
implement Administrator permissions for deviceAttributes and webhooks. The first try.

AntonPuko committed:
implement deviceKeyDatabase repo

AntonPuko committed:
throw a error if device is already claimed by the user

jlkalberer committed:
Merge pull request #158 from AntonPuko/addDoubleClaimDeviceThrows

throw an error if device is already claimed by the user

jlkalberer committed:
Merge pull request #159 from AntonPuko/dbkey

implement deviceKeyDatabase repo

jlkalberer committed:
Updates and improvements from setting up particle-collider

jlkalberer committed:
Fixing tests.. again.

jlkalberer committed:
Merge branch 'dev' of github.com:Brewskey/spark-server into dev

jlkalberer committed:
Fixing logger so it stringifies objects correctly.

AntonPuko committed:
made deviceKeyRepository consistent with IBaseRepository

jlkalberer committed:
Rebuild

jlkalberer committed:
Fixing timeouts for loading large amounts of data.

jlkalberer committed:
Fixing access token so it doesn't expire so soon

jlkalberer committed:
Merge pull request #161 from AntonPuko/deviceKeyFix

made deviceKeyRepository consistent with IBaseRepository

AntonPuko committed:
get token with Ouath hacky code.

AntonPuko committed:
move oauthServer initialization from RouteConfig to defaultBindings

jlkalberer committed:
Merge pull request #154 from AntonPuko/permissions

implement Administrator permissions first try

AntonPuko committed:
rebuild

AntonPuko committed:
change arg from id to query in remove db method. fix deleteById name

AntonPuko committed:
rename getByID

AntonPuko committed:
fix OauthModel and `accessTokenExpiresAt` must be a Date instance' error.

AntonPuko committed:
don't replace _id with deviceID in database repos, but use deviceID as query.

jlkalberer committed:
Merge pull request #166 from AntonPuko/fixModel

fix OauthModel and `accessTokenExpiresAt` must be a Date instance' er…

AntonPuko committed:
clean Flow types

jlkalberer committed:
Merge pull request #167 from AntonPuko/reposFixes

Repos fixes

jlkalberer committed:
Allow passing in pre-created express app to do custom routing.

AntonPuko committed:
add deviceKeys migration to migrateScript

jlkalberer committed:
Merge branch 'dev' of github.com:Brewskey/spark-server into dev

jlkalberer committed:
Merge pull request #168 from AntonPuko/migrationScript

add deviceKeys migration to migrateScript

jlkalberer committed:
Create azure-mongodb.js

AntonPuko committed:
disable eslint in azure example

AntonPuko committed:
fix package-lock.json

jlkalberer committed:
Merge pull request #174 from AntonPuko/fix-package-lock

Fix package lock

Commit:
Added a Docker Build Script and Explanation

jlkalberer committed:
Merge pull request #176 from keatec/dev

Added a Docker Build Script and Explanation

jlkalberer committed:
Added `ping` route

AntonPuko committed:
use _eventPublisher.publishAndListenForResponse instead coupling of deviceServer. fix tests, fix raise your hand issue.

haeferer committed:
Merge pull request #1 from Brewskey/dev

Update Changes from Brewskey

Commit:
added setting.json so vscode works with flow

jlkalberer committed:
Merge pull request #179 from keatec/dev

add configuration to be compatible with VisualStudioCode

jlkalberer committed:
Merge pull request #177 from AntonPuko/messaging

publishAndListenForResponse

AntonPuko committed:
fix var name nit

jlkalberer committed:
Flow fixes

jlkalberer committed:
Merge branch 'dev' of github.com:Brewskey/spark-server into dev

jlkalberer committed:
Removing ursa

jlkalberer committed:
Allow custom template replacement on webhooks.

haeferer committed:
Merge pull request #2 from Brewskey/dev

Merge Changes from Brewskey

Commit:
Added Image to Docker HUB, remove Node-GYP from Build Process

Commit:
Merge branch 'dev' of https://github.com/keatec/spark-server into dev

jlkalberer committed:
Merge pull request #182 from keatec/dev

Update DockerFile and Instructions

Commit:
first text

Commit:
Added ReadMe on Compile with docker

Commit:
Patch

jlkalberer committed:
Merge pull request #186 from keatec/dev

Added Instructions under /examples to compile local with docker

straccio committed:
Pre fork from Brewskey

straccio committed:
Merge remote-tracking branch https://github.com/Brewskey/spark-server 'upstream/master'

straccio committed:
Merge remote-tracking branch 'upstream/dev'

straccio committed:
Package for debugging purpose

straccio committed:
Corrected get devices that now return "connected" attribute correctly

straccio committed:
Removed h5.coap and added script for debugging

straccio committed:
package-lock.json

jlkalberer committed:
Merge pull request #187 from straccio/master

Problem with get devices

AntonPuko committed:
change update to updateByID and use as PATH instead PUT

jlkalberer committed:
Merge pull request #188 from AntonPuko/repoUpdateByID

change update to updateByID and use it as PATH instead PUT

haeferer committed:
Merge pull request #3 from Brewskey/dev

Added Commits from Brewskey

AntonPuko committed:
use getAttributes for devices from device-server.

jlkalberer committed:
Merge pull request #190 from AntonPuko/attr2

use getAttributes for devices from device-server.

AntonPuko committed:
fix key format. and build some not built files

jlkalberer committed:
Merge pull request #192 from AntonPuko/fixKeyFormat

fix key format. and build some not built files

haeferer committed:
Merge pull request #4 from Brewskey/dev

Added Changings from Brewskey

jlkalberer committed:
Merge pull request #193 from keatec/dev

Merge pull request #3 from Brewskey/dev

jlkalberer committed:
Updated firmware endpoint so you can send tinker from the cli

jlkalberer committed:
Updated logger so it has a timestamp and allow overrides via constitute

closes #195

jlkalberer committed:
Adding electron support

jlkalberer committed:
Updating for ecc

jlkalberer committed:
Added electron support. closes #191

AntonPuko committed:
implement NeDb

jlkalberer committed:
Merge pull request #197 from AntonPuko/nedb

implement NeDb

jlkalberer committed:
Logger fixes + adding prebuilt files

haeferer committed:
Merge pull request #5 from Brewskey/dev

Adding Changings from

haeferer committed:
Added new Logger Structur

haeferer committed:
Implemented Logger Constitute

haeferer committed:
Added suggestions from jlkalberer

haeferer committed:
added suggestions

haeferer committed:
Added changes from jlkalberer

haeferer committed:
Add Logger Initialization

haeferer committed:
Add extraline to types.js

haeferer committed:
Rename DefaultLogger

haeferer committed:
Rename DefaultLogger final

Commit:
added git:clean to package.json to checkout all files from ./dist

haeferer committed:
remove git:clean, doesnt make sense

haeferer committed:
DefaultLogger now matches Filename

jlkalberer committed:
Merge pull request #199 from keatec/dev

Logger Constitution

haeferer committed:
Merge pull request #8 from Brewskey/dev

Merge pull request #199 from keatec/dev

AntonPuko committed:
fix deps

AntonPuko committed:
Merge pull request #208 from AntonPuko/fixDeps

fix deps

haeferer committed:
Merge pull request #9 from Brewskey/dev

Merge from brewskey

haeferer committed:
Disable Cache on AVA, to tests can be run without write permission to node_modules

jlkalberer committed:
Merge pull request #211 from keatec/dev

Added --no-cache to AVA in package.json

AntonPuko committed:
update MongoDb so it waits for connection ready.

AntonPuko committed:
delete TingoDB

AntonPuko committed:
update migrate-script

AntonPuko committed:
update azure-mongo example

jlkalberer committed:
Merge pull request #213 from AntonPuko/updateDbs

Update dbs

AntonPuko committed:
filter broadcasted events in webhookmanager

jlkalberer committed:
Merge pull request #214 from AntonPuko/eventProvider2

filter broadcasted events in webhookmanager

AntonPuko committed:
add eventProvider example

jlkalberer committed:
Merge pull request #215 from AntonPuko/eventProviderExample

add eventProvider example

haeferer committed:
Merge pull request #12 from Brewskey/dev

Added Changes from Brewskey

haeferer committed:
Added npm functions for example build and run, changed eventProvider sample

Commit:
Changed sample

Commit:
changed ignore on npm build:examples

Commit:
patched sample to working code

AntonPuko committed:
add prettier, add gitAttributes to hide dist in commit diffs

jlkalberer committed:
Merge pull request #223 from AntonPuko/prettier

add prettier, add gitAttributes to suppress dist in commit diffs

haeferer committed:
Changed Infos in package.json

haeferer committed:
Merge pull request #18 from Brewskey/dev

Copy changes from brewskey

haeferer committed:
Added Bunyan Logger

haeferer committed:
added last NIT from last PR

haeferer committed:
Added Changes from jlkalberer

haeferer committed:
Added Changes AntonPuko

haeferer committed:
Added WebHookLogger Log Line

AntonPuko committed:
remove gitAdd from npm scripts, its lint-staged own command

AntonPuko committed:
Merge pull request #226 from keatec/bunyanlog

Implementation of Bunyan Logger

AntonPuko committed:
update package-lock.json, closes https://github.com/Brewskey/spark-server/issues/229

AntonPuko committed:
remove wrong logger.warn in webhookManager

AntonPuko committed:
remove wrong logger.warn in webhookManager

AntonPuko committed:
update package-lock.json, closes https://github.com/Brewskey/spark-server/issues/234

AntonPuko committed:
Merge branch 'dev' of github.com:Brewskey/spark-server into dev

AntonPuko committed:
update migrateScript, so it saves dates correctly.

jlkalberer committed:
Merge pull request #235 from AntonPuko/migrateScriptFix

update migrateScript, so it saves dates correctly.

straccio committed:
Packages update.

AntonPuko committed:
passing not json event.data as sting for webhooks as form-ulr-encoded

jlkalberer committed:
Merge pull request #236 from AntonPuko/webhookManagerFix

passing not json event.data as sting for webhooks as form-ulr-encoded

straccio committed:
Merge remote-tracking branch 'upstream/dev'

AntonPuko committed:
update build

AntonPuko committed:
stringify device variables

jlkalberer committed:
Merge pull request #238 from AntonPuko/stringifyDeviceVariabes

stringify device variables

straccio committed:
Merge remote-tracking branch 'upstream/dev'

jlkalberer committed:
try/catch around device attributes as JSON deserializer was breaking on existing devices.

jlkalberer committed:
Update to newer version of spark protocol

AntonPuko committed:
build deviceAttributesDatabaseRepository.js

straccio committed:
Merge remote-tracking branch 'upstream/dev'

jlkalberer committed:
Starting work on products API.
Still need to add in device endpoint and update the firmware manager so it will auto-update devices.

AntonPuko committed:
fix adminUser creation

jlkalberer committed:
Merge pull request #240 from AntonPuko/adminFix

fix adminUser creation

jlkalberer committed:
Finished API for managing firmware/devices/products.

Now we need to actually add code for flashing the correct firmware when the device connects.

jlkalberer committed:
Merge branch 'dev' of github.com:Brewskey/spark-server into dev

jlkalberer committed:
Added changes so that when a product firmware is updated, it sends an event to the device server to start updating all connected devices.

jlkalberer committed:
Rebuilt dist

jlkalberer committed:
Merge branch 'dev' of github.com:Brewskey/spark-server into dev

# Conflicts:
#	dist/repository/DeviceAttributeDatabaseRepository.js

jlkalberer committed:
Fixed DB so it fetches correctly for mongo.

AntonPuko committed:
fix variables loss on update device name.

jlkalberer committed:
Merge pull request #243 from AntonPuko/fixUndefinedOnUpdateAttr

fix variables loss on update device name.

AntonPuko committed:
remove console.logs; add webhookLoggerStub

AntonPuko committed:
add bunyan to start scripts

AntonPuko committed:
add DeviceManager to ProductsController bindings

AntonPuko committed:
provide webhookData as querystring for GET requests

AntonPuko committed:
update package-lock.json to get last spark-protocol changes.

AntonPuko committed:
remove comments

AntonPuko committed:
add deviceManager assignment in productsController, closes https://github.com/Brewskey/spark-server/issues/245

jlkalberer committed:
Merge pull request #246 from AntonPuko/fixBugs

Fix bugs

AntonPuko committed:
update package-lock.json for spark-protocol change: https://github.com/Brewskey/spark-protocol/commit/3472b1a3d312b1d50d537a8a28ce877f5e2c82b3

closes https://github.com/Brewskey/spark-server/issues/244

AntonPuko committed:
update package-lock.json for spark-protocol change: https://github.com/Brewskey/spark-protocol/commit/b5ede6dbaa5cc16e6f4705d2dec930aacd64d089

AntonPuko committed:
update package-lock.json for spark-protocol change: https://github.com/Brewskey/spark-protocol/commit/2e1dde37210078221e22a085e0e8a89d642df15d

AntonPuko committed:
update package-lock.json for spark-protocol change: https://github.com/Brewskey/spark-protocol/commit/6e83a4613f953485d4491d91bf5907adc5fe2d5d

AntonPuko committed:
product api fixes

jlkalberer committed:
Merge pull request #248 from AntonPuko/fixProducts

product api fixes

AntonPuko committed:
revert back casting product_id to string on production creation, use helper function for getting numeric productID from IDOrSlug

AntonPuko committed:
add equals to 0 check for right platform_id filtering.

AntonPuko committed:
fix _formatProduct in ProductDatabaseRepository

AntonPuko committed:
add parseInt for firmware version comes from route params.

AntonPuko committed:
formatDeviceAttributes, add product_id to returned product devices props

AntonPuko committed:
build files

AntonPuko committed:
use isNaN in getByIDOrSlug

jlkalberer committed:
Merge pull request #249 from AntonPuko/productFixes

Product api fixes 2

AntonPuko committed:
fix flashProductDevices method, fix csv parsing

jlkalberer committed:
Merge pull request #250 from AntonPuko/productFixes3

fix flashProductDevices method, fix csv parsing

AntonPuko committed:
save product_id instead product.id in productDevice

AntonPuko committed:
cast productFirmware buffer when save to/ get from db.

jlkalberer committed:
Merge pull request #251 from AntonPuko/productFixes4

Product fixes4

AntonPuko committed:
update flow-bin, fix some product related flow errors, there are still some

AntonPuko committed:
fix bugs, add temporary hack for getting right buffer from different dbs

AntonPuko committed:
add devicePing endpoint

jlkalberer committed:
Merge pull request #253 from AntonPuko/pingDevice

add pingDevice endpoint

jlkalberer committed:
Merge pull request #252 from AntonPuko/productFixes5

fix bugs, add temporary hack for getting right buffer from different dbs

straccio committed:
Create http server with timeout 0

Usefull for wathcing subscriptions (events)

straccio committed:
Merge remote-tracking branch 'upstream/dev'

I have disabled some tests on WebhookManager and EventsController

straccio committed:
Build

AntonPuko committed:
disable userFilter for admin in eventsController

jlkalberer committed:
Merge pull request #254 from AntonPuko/disableUserFilterForAdmin

disable userFilter for admin in eventsController

AntonPuko committed:
return keep alive for event stream

jlkalberer committed:
Merge pull request #255 from AntonPuko/eventskeepalive

return keep alive for event stream

straccio committed:
Merge remote-tracking branch 'upstream/dev'
